Mathlib/RingTheory/AlgebraicIndependent/TranscendenceBasis.lean

```diff
@@ -124,3 +208,302 @@ theorem IsTranscendenceBasis.isAlgebraic_field {F E : Type*} {x : ι → E}
+    lift.{u'} #ι = lift.{u} #ι' := by
+  rw [← lift_inj.{_, w}, lift_lift, lift_lift, ← lift_lift.{w, u'}, hx.lift_cardinalMk_eq_trdeg,
+    ← lift_lift.{w, u}, hy.lift_cardinalMk_eq_trdeg, lift_lift, lift_lift]
+
+@[stacks 030F] theorem cardinalMk_eq {ι' : Type u} {y : ι' → A}
+    (hx : IsTranscendenceBasis R x) (hy : IsTranscendenceBasis R y) :
+    #ι = #ι' := by
+  rw [← lift_id #ι, lift_cardinalMk_eq hx hy, lift_id]
+
+end IsTranscendenceBasis
+
+-- TODO: generalize to Nontrivial S
+@[simp]
+theorem MvPolynomial.trdeg_of_isDomain [IsDomain S] : trdeg S (MvPolynomial ι S) = lift.{v} #ι := by
+  have := (IsTranscendenceBasis.mvPolynomial ι S).lift_cardinalMk_eq_trdeg.symm
+  rwa [lift_id', ← lift_lift.{u}, lift_id] at this
+
+-- TODO: generalize to Nontrivial R
+@[simp]
+theorem Polynomial.trdeg_of_isDomain [IsDomain R] : trdeg R (Polynomial R) = 1 := by
+  simpa using (IsTranscendenceBasis.polynomial Unit R).lift_cardinalMk_eq_trdeg.symm
+
+-- TODO: generalize to Nontrivial S
+theorem trdeg_lt_aleph0 [IsDomain R] [fin : FiniteType R S] : trdeg R S < ℵ₀ :=
+  have ⟨n, f, surj⟩ := FiniteType.iff_quotient_mvPolynomial''.mp fin
+  lift_lt.mp <| (lift_trdeg_le_of_surjective f surj).trans_lt <| by
+    simpa using Cardinal.nat_lt_aleph0 _
+
+namespace Algebra.IsAlgebraic
+
+variable (R x) (s : Set A)
+
+variable [NoZeroDivisors A]
+
+lemma isDomain_of_adjoin_range [Algebra.IsAlgebraic (adjoin R s) A] : IsDomain A :=
+  have := Algebra.IsAlgebraic.nontrivial (adjoin R s) A
+  (isDomain_iff_noZeroDivisors_and_nontrivial _).mpr
+    ⟨‹_›, (Subtype.val_injective (p := (· ∈ adjoin R s))).nontrivial⟩
+
+theorem trdeg_le_cardinalMk [alg : Algebra.IsAlgebraic (adjoin R s) A] : trdeg R A ≤ #s := by
+  by_cases h : Injective (algebraMap R A)
+  on_goal 2 => simp [trdeg_eq_zero_of_not_injective h]
+  have := isDomain_of_adjoin_range R s
+  have := (faithfulSMul_iff_algebraMap_injective R A).mpr h
+  rw [← matroid_spanning_iff, ← matroid_cRank_eq] at *
+  exact alg.cRank_le_cardinalMk
+
+variable [FaithfulSMul R A]
+
+theorem isTranscendenceBasis_of_lift_le_trdeg_of_finite
+    [Finite ι] [alg : Algebra.IsAlgebraic (adjoin R (range x)) A]
+    (le : lift.{w} #ι ≤ lift.{u} (trdeg R A)) : IsTranscendenceBasis R x := by
+  have ⟨_, h⟩ := lift_mk_le'.mp (le.trans <| lift_le.mpr <| trdeg_le_cardinalMk R (range x))
+  have := surjective_onto_range.bijective_of_nat_card_le (Nat.card_le_card_of_injective _ h)
+  refine .of_subtype_range (fun _ _ ↦ (this.1 <| Subtype.ext ·)) ?_
+  have := isDomain_of_adjoin_range R (range x)
+  rw [← matroid_spanning_iff, ← matroid_cRank_eq] at *
+  exact alg.isBase_of_le_cRank_of_finite (lift_le.mp <| mk_range_le_lift.trans le) (finite_range x)
+
+theorem isTranscendenceBasis_of_le_trdeg_of_finite {ι : Type w} [Finite ι] (x : ι → A)
+    [Algebra.IsAlgebraic (adjoin R (range x)) A] (le : #ι ≤ trdeg R A) :
+    IsTranscendenceBasis R x :=
+  isTranscendenceBasis_of_lift_le_trdeg_of_finite R x (by rwa [lift_id, lift_id])
```

Comment:
How hard is it to get the `IntermediateField.adjoin` version of these lemmas?

Reply:
see #24004